migration-engine/ARCHITECTURE.md

```diff
@@ -117,6 +118,41 @@ On a high level, migrate deploy _exclusively_:
   looking at the `_prisma_migrations` table in the target database.
 - Run the migrations that have not been applied yet, in chronological order.
 
+`prisma migrate deploy` is the command meant to be used for _unattended_
+migrations (as part of CI). As such, it should be as reliable, predictable and
+deterministic as possible.
+
+### Why does `prisma migrate deploy` not warn when a migration is in the migrations table but not in the migrations folder?
+
+We don't want to warn if an already applied migration is missing from the
+migration directory because it would prevent squashing migrations and
+deployment from not-properly-rebased branches.
```

Comment:
OT: Is squashing migrationssomething we support with a workflow or a manual thing?

Reply:
It's something we don't communicate in the docs, but there is no reason we shouldn't, it works and we have tests in the migration engine checking that it does.

Reply:
Product never picked it up as a required feature, but the general idea was to not communicate about it until it's gotten product design effort put into it.